flink-runtime/src/main/java/org/apache/flink/runtime/state/AbstractKeyedStateBackend.java

```diff
@@ -127,7 +127,9 @@ public AbstractKeyedStateBackend(
                 numberOfKeyGroups >= 1, "NumberOfKeyGroups must be a positive number");
         Preconditions.checkArgument(
                 numberOfKeyGroups >= keyGroupRange.getNumberOfKeyGroups(),
-                "The total number of key groups must be at least the number in the key group range assigned to this backend");
+                "The total number of key groups must be at least the number in the key group range assigned to this backend %s %s",
```

Comment:
Hmm, maybe even more specific: 
"The total number of key groups must be at least the number in the key group range assigned to this backend. The total number of key groups=%s ; the number in the key group range=%s"

Or something similar?

Reply:
Good idea!